gitlab/pkg/apis/apis.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2020 The TriggerMesh Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Generate deepcopy for apis
+//go:generate go run ../../vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go -O zz_generated.deepcopy -i ./... -h ../../hack/boilerplate.go.txt
+
+// Package apis contains Kubernetes API groups.
+package apis
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// AddToSchemes may be used to add all resources defined in the project to a Scheme
+var AddToSchemes runtime.SchemeBuilder
+
+// AddToScheme adds all Resources to the Scheme
+func AddToScheme(s *runtime.Scheme) error {
```

Comment:
This is usually done at the API level (`v1xyz`) ([example1](https://github.com/knative/eventing-contrib/blob/release-0.12/kafka/source/pkg/apis/sources/v1alpha1/register.go#L47-L60), [example2](https://github.com/knative/eventing/blob/release-0.12/pkg/apis/sources/v1alpha1/register.go#L40-L57)). What is the reason for registering _all_ GitLabSources versions at once?

Reply:
up